piptools/writer.py

```diff
@@ -159,23 +163,25 @@ def _iter_lines(
         packages = {r for r in results if r.name not in UNSAFE_PACKAGES}
 
         packages = sorted(packages, key=self._sort_key)
```

Comment:
Shouldn't this line be also under `if packages`?

Reply:
Indeed it should. No point in sorting nothing.